src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java

```diff
@@ -282,6 +282,6 @@ public static boolean isNotEmpty( String str )
 
     public static boolean isEmpty( String str )
     {
-        return ( str == null || str.trim().length() == 0 );
+        return ( str == null || str.length() == 0 );
```

Comment:
This is fatally UGLY. P-U has StringUltils#isEmtpy(). Why not remove and use that?

Reply:
It is easier to spot the difference when keeping the other conditions as is. I would rather cleanup in a subsequent commit. WDYT?

Reply:
Agreed. Leave a comment about removal.

Reply:
This is a public method, so probably this needs deprecation before removing it finally.

Reply:
Makes sense. The impl can be exchanged for StringUtils for consistency reasons.

Reply:
I added https://github.com/codehaus-plexus/plexus-utils/issues/215 to track that.